File: internal/cocktailingredient.go
package internal

type CocktailIngredient struct {
	CocktailID   int
	IngredientID int
	Quantity     int
	Unit         Unit
}

func NewCocktailIngredient(cocktailID, ingredientID, quantity int, unit Unit) CocktailIngredient {
	return CocktailIngredient{
		CocktailID:   cocktailID,
		IngredientID: ingredientID,
		Quantity:     quantity,
		Unit:         unit,
	}
}

File: SipSavy.Data/Domain/CocktailIngredient.cs
namespace SipSavy.Data.Domain;

public sealed class CocktailIngredient
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public float Quantity { get; set; }
    public Unit Unit { get; set; } = Unit.None;

    public int CocktailId { get; set; }
    public Cocktail Cocktail { get; set; } = null!;
}